### PR TITLE
sentry: port sentry support (bug 1873221)

### PR DESCRIPTION
**Pull request by zzzeid**

### CODE REVIEW

File: src/lando/main/sentry.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+logger = logging.getLogger(__name__)
+
+
+def init_sentry():
+    """Initialize Sentry integration for remote environments."""
+    if not settings.ENVIRONMENT.is_remote:
+        raise ValueError(
+            "Attempting to initialize Sentry in non-remote environment "
+            f"({settings.ENVIRONMENT})"
+        )
```
Comment (cgsheeh) [inline src/lando/main/sentry.py:18]: Nit: we also check this in the caller, it can be removed.

File: src/lando/main/sentry.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    sentry_sdk.init(
+        dsn=settings.SENTRY_DSN,
+        environment=settings.ENVIRONMENT,
+        integrations=[DjangoIntegration()],
+        traces_sample_rate=1.0,
+        release=version,
+    )
```
Comment (cgsheeh) [inline src/lando/main/sentry.py:25]: In the previous implementation we have a log entry to indicate the status of Sentry, should we carry that forward?

**Review by cgsheeh — APPROVED** [review]

File: src/lando/main/sentry.py (hunk 2 above)
Comment (zzzeid) [reply src/lando/main/sentry.py:25]: I don't think that's necessary here, don't think I've ever looked for those logs. Easy enough to re-add if we ever feel the need to do that.

File: src/lando/main/sentry.py (hunk 1 above)
Comment (zzzeid) [reply src/lando/main/sentry.py:18]: Changed up the implementation slightly to simplify this.

**Review by cgsheeh — APPROVED** [review]

**Pull request merged**